In [21]:
 import pandas as pd
 import numpy as np
 import seaborn as sns
 import matplotlib.pyplot as plt
 from scipy import stats

#Q1) 
Null hypothesis as Ho, u1=u2 (Where the diameters of cutlets of 2 units as equal)
Alternate hypothesis as Ha, u1=/u2 (where the diameters of cutlets of 2 units as not equal)

So, we will be using 2 sample 2 tail test

In [2]:
from google.colab import files
uploaded=files.upload()

Saving Cutlets.csv to Cutlets.csv


In [3]:
df=pd.read_csv("Cutlets.csv")

In [4]:
df

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [15]:
Unit_A=pd.Series(df.iloc[:,0])

In [16]:
Unit_A

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [17]:
Unit_B=pd.Series(df.iloc[:,1])

In [18]:
Unit_B

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [22]:
p_value=stats.ttest_ind(Unit_A,Unit_B)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

P-Value=0.4722395 which is greater than 0.05,
p_value=0.4722395>0.05.
So, we **fail to reject Null Hypothesis**. That is there is no difference in diameters of the cutlets of 2 units.

#Q2)
For analysis of variance between more than 2 samples we use Anova ftest.
Assumes Null hypothesis Ho as all TAT samples has same mean.
So, Assume alternate hypothesis Ha as TAT samples mean is different.

In [28]:
from google.colab import files
uploaded=files.upload()

Saving LabTAT.csv to LabTAT.csv


In [29]:
labTAT=pd.read_csv("LabTAT.csv")

In [30]:
labTAT.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [32]:
p_value=stats.f_oneway(labTAT.iloc[:,0],labTAT.iloc[:,1],labTAT.iloc[:,2],labTAT.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [33]:
p_value[1]

2.1156708949992414e-57

P-Value < 0.05 so we **accept Alternate Hypothesis**.
**There is a difference in average TAT among different laboratories**

#Q3)
Assume Null hypothesis Ho as male-female buyer rations are similar across regions.
Assume Alternate hypothesis Ha as male-female buyer rations are not similar across the regions

In [34]:
from google.colab import files
uploaded=files.upload()

Saving BuyerRatio (1).csv to BuyerRatio (1).csv


In [37]:
br=pd.read_csv("BuyerRatio (1).csv")

In [39]:
br

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [40]:
from scipy.stats import chi2_contingency

In [42]:
#Creating an array
br1=np.array([[50,142,131,70],[435,1523,1356,750]])
br1

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [43]:
chi2_contingency(br1)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

Output will display (chi stats value, P-Value, Degree of Freedom, expected observations)

P-Value is 0.66031>0.05 So, we **fail to reject null hypothesis**.
That is male-female buyer rations are **sililar** across the regions

#Q4)
Assume Null hypothesis Ho as customer order forms does not vary across centres.
& assume Alternate hypothesis Ha as customer order forms varies across the centres.

In [44]:
from google.colab import files
uploaded=files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


In [45]:
cof=pd.read_csv("Costomer+OrderForm.csv")
cof

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [46]:
cof["Phillippines"].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [47]:
cof["Indonesia"].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [48]:
cof["Malta"].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [49]:
cof["India"].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [51]:
#Creating an array
ar=np.array([[271,267,269,280],[29,33,31,20]])
ar

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [52]:
chi2_contingency(ar)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

Output will display (chi stats value, P-Value, Degree of Freedom, expected observations)

P-Value is 0.277102 which is >0.05 So, we **Accept Null Hypothesis**.
That is customer order forms does not vary among the centres.